x/stake/keeper.go

```diff
@@ -469,3 +469,67 @@ func (k Keeper) setPool(ctx sdk.Context, p Pool) {
 	store.Set(PoolKey, b)
 	k.gs = Pool{} // clear the cache
 }
+
+//__________________________________________________________________________
+
+// Implements ValidatorSetKeeper
+
+var _ sdk.ValidatorSetKeeper = Keeper{}
+
+func (k Keeper) Validators(ctx sdk.Context) []*sdk.Validator {
+	vals := k.GetValidators(ctx)
+	res := make([]*sdk.Validator, len(vals))
+
+	for i, val := range vals {
```

Comment:
We shouldn't use loops in this keeper, as well, this functionality already exists in this keeper without using loops

Reply:
`GetValidators` returns `stake.Validator` which is different from `sdk.Validator`. The loop is for type conversion. We can make `sdk.Validator` an interface and make `stake.Validator` implement it.